### ISSUE TITLE
Pass configuration files to plugins independently

### ISSUE THREAD

**Issue by alexander-yevsyukov** [SpineEventEngine/ProtoData#198]
We need to simplify the way settings are passed to plugins and renderers in ProtoData. Also, the knowledge of those settings should be made local to plugins and renderers as much as possible.

### The problem
Currently only one ProtoData plugin, namely `io.spine.validation.java.JavaValidationPlugin` from the Validation library, uses the feature of loading configuration files passed to ProtoData plugins.

Configuration settings for the `JavaValidationPlug` are written by `io.spine.tools.mc.java.gradle.plugins.GenerateProtoDataConfig` task on disk as binary representation of `io.spine.validation.ValidationConfig` Protobuf message. See `GenerateProtoDataConfig.writeFile()` for details.

`ValidationConfig` is loaded by `io.spine.validation.RequiredIdOptionPolicy` and `io.spine.validation.RequiredIdPatternPolicy` via `configAs(ValidationConfig.class)` calls. The `Policy` class assumes that there is only one configuration passed to ProtoData. See `io.spine.protodata.config.ConfigurationQuerying` interface for the following line:

```kotlin
val config = configurations.theOnly()
```

A `Renderer` in ProtoData also assumes `theOnly` configuration because it inherits the same `ConfigurationQuerying` interface as `Policy` classes do, and make it `final` likewise.

Therefore, currently ProtoData can only accept one configuration file, which is a Protobuf message in one of the serialization forms. In order for plugins or renderers to get their settings, we need to have an "umbrella" Protobuf message which has **all the settings for all the plugins**. And plugins have to depend on this "umbrella" type just to get their settings.

#### Side problem

McJava has significant part of code which deals with "gymnastics" of transforming extensions of a Gradle plugin to Protobuf messages so that we can pass them to ProtoData. It would be good to have a common approach which is based on exposing Protobuf Kotlin DSL in a Gradle build.

#### Write and load configuration for each ProtoData plugin independently
We can use the same `build/protodata-config` directly, but using plugin-based format for file names:
```
io.spine.validation.java.JavaValidationPlugin-main.bin
io.spine.validation.java.JavaValidationPlugin-testFixtures.bin
io.spine.tools.mc.annotation.ApiAnnotationsPlugin-test.json
```

### Expose Protobuf message types for ProtoData plugin configuration directly
Now we have `io.spine.tools.mc.java.gradle.codegen` package that has a hierarchy of `Config` classes that serve for adaptation between types used in Gradle Extension of McJava and corresponding Protobuf message types used for passing settings to `protoc` plugins or ProtoData plugins. Let's try to make it more obvious and simpler to use by only having separate Protobuf message types added to a "root" settings object dynamically using Gradle API.

### Subtasks
- [ ] Fail if more than one file (having different formats) are passed to the same consumer.

**Comment by alexander-yevsyukov** [comment]
@armiol, please have a look at my today's findings on passing settings to new API annotations impl. Let's discuss later today.

**Comment by alexander-yevsyukov** [comment]
Resolved by #205.
